File: backend/app/models/project.py
"""
Project entry model — parsed from PROJECT_SWEEP_SUMMARIES.md.
"""
from pydantic import BaseModel
from typing import List, Optional


class ProjectEntry(BaseModel):
    id: str
    name: str
    type: str
    summary: str
    tech_stack: List[str] = []
    key_features: List[str] = []
    resume_value_bullets: List[str] = []
    domains: List[str] = []
    lines_of_code: Optional[int] = None
    source_section: str = ""

File: frontend/src/types/project.ts
/**
 * TypeScript interfaces mirroring backend/app/models/project.py
 */

export interface ProjectEntry {
  id: string;
  name: string;
  type: string;
  summary: string;
  tech_stack: string[];
  key_features: string[];
  resume_value_bullets: string[];
  domains: string[];
  lines_of_code?: number;
  source_section: string;
}
